js/sign.js

```diff
@@ -2,100 +2,200 @@ const useremail = document.querySelector('#useremail')
+export const password = document.querySelector('#password');
 const signInputPassword = document.querySelector('.sign-input-password');
 
-function passwordTypeChacking (password){
+// 비밀번호 타입이 잘못되었는지 확인하는 함수
+function checkPasswordType (password){
+  //이메일 형식이 알파벳+숫자@알파벳+숫자.알파벳+숫자 형식이 아닐경우			
   const exptext = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
   if(exptext.test(password) === false){
-  //이메일 형식이 알파벳+숫자@알파벳+숫자.알파벳+숫자 형식이 아닐경우			
-  errPaint('비밀번호는 영문,숫자 조합 8자 이상 입력해주세요.', signInputPassword)
+  paintErr('비밀번호는 영문,숫자 조합 8자 이상 입력해주세요.', signInputPassword)
   return false;
   }
+  paintErrLine(password);
 }
 
-// 비밀번호 이상 유무 체크
-function chackingPassword () {
+
+function checkPassword () {
   // 에러코드 중복 방지
-  if (password.parentNode.childNodes[7]){
-    password.parentNode.childNodes[7].remove();
+  if (signInputPassword.lastChild.tagName === "P"){
+    signInputPassword.lastChild.remove();
   }  
   if (!password.value){
-    errPaint('비밀번호를 입력해주세요', signInputPassword);
+    paintErr('비밀번호를 입력해주세요', signInputPassword);
+    paintErrLine(password);
+  } else if (checkPasswordType(password.value)){
+    paintErrLine(password);
   } else{
-    passwordTypeChacking(password.value);
-  } 
+    removeErrLine(password);
+  }
 }
 
-//비밀번호 재확인
-const password2 = document.querySelector('#password2');
+
+export const password2 = document.querySelector('#password2');
 const signInputPassword2 = document.querySelector('.sign-input-password2');
 
-function doubleCheakingPassword(){
+function doubleCheckPassword(){
   // 에러코드 중복 방지
-  if (password2.parentNode.childNodes[7]){
-    password2.parentNode.childNodes[7].remove();
+  if (signInputPassword2.lastChild.tagName === "P"){
+    signInputPassword2.lastChild.remove();
   }
   if (password.value !== password2.value){
-    errPaint('비밀번호가 일치하지 않아요', signInputPassword2);
+    paintErr('비밀번호가 일치하지 않아요', signInputPassword2);
+    paintErrLine(password);
   }
 }
 
 // 로그인, 회원가입
 const signForm = document.querySelector('.sign-middle');
 
-function submitForm (e) {
+async function submitForm (e) {
   e.preventDefault();
-  if (useremail.value && (password.value || password2.value)){
-    if (signInputEmail.lastChild.tagName !== "P" && (signInputPassword.lastChild.tagName !== "P" || signInputPassword2.lastChild !== "P")){
-      location.href = "/folder";
-      return; 
+  try {
+    const response = await fetch("https://bootcamp-api.codeit.kr/api/sign-in", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        "email": useremail.value,
+        "password": password.value,
+      }),
+    })
+    if (response.status === 200){
+      if (useremail.value === 'test@codeit.com' && password.value === 'sprint101'){
+        location.href = "/folder";
+        return;
+      }
     }
+  }catch{
+    checkLogin();
+  };
+
+  try {
+    const response = await fetch("https://bootcamp-api.codeit.kr//api/sign-up", {
```

Comment:
<img width="702" alt="image" src="https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/60066472/779f28fc-24f0-4cc0-9447-87b71058f285">


>  [x] [기본]유효한 회원가입 형식의 경우 “/api/sign-up” POST 요청하고 성공 응답을 받으면 “/folder”로 이동하나요?

PR을 올리기 전에 요구사항의 기능을 직접 최소 1번 테스트 해주시고, 만약 막혔더라도 어디까지 시도해보셨는지 같이 공유해주시면 더 좋을 것 같습니다 :)


```suggestion
    const response = await fetch("https://bootcamp-api.codeit.kr/api/sign-up", {
```